101-datastore-create-blob/azuredeploy.json

```diff
@@ -0,0 +1,110 @@
+{
+	"$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
+		"workspaceName": {
+			"type": "string",
+			"metadata": {
+				"description": "Specifies the name of the Azure Machine Learning workspace which will hold this datastore target."
+			}
+		},
+		"datastoreName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the datastore, case insensitive, can only contain alphanumeric characters and underscore"
+			}
+		},
+		"accountName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the storage account."
+			}
+		},
+		"containerName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the azure blob container."
+			}
+		},
+		"storageAccountSubscriptionId": {
+			"type": "string",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : The subscription ID of the storage account."
+			}
+		},
+		"storageAccountResourceGroup": {
+			"type": "string",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : The resource group of the storage account."
+			}
+		},
+		"accountKey": {
+			"type": "securestring",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : A storage account key, defaults to None. Skip this if SAS token authentication is used."
+			}
+		},
+		"sasToken": {
+			"type": "securestring",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : An account SAS token, defaults to None. Skip this if storage account key authentication is used."
+			}
+		},
+		"endPoint": {
+			"type": "string",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : The endpoint of the storage account. Defaults to core.windows.net"
+			}
+		},
+		"protocol": {
+			"type": "string",
+			"defaultValue":"",
+			"metadata": {
+				"description": "Optional : Protocol to use to connect to the blob container. If not specified, defaults to https."
```

Comment:
what are the options for this?  Would we ever not want to use https on Azure Storage?

Reply:
Don't think so, we expose this to customer in our other products like [CLI ](https://docs.microsoft.com/en-us/cli/azure/ext/azure-cli-ml/ml/datastore?view=azure-cli-latest#ext-azure-cli-ml-az-ml-datastore-attach-blob)so exposing it here too.

Reply:
Let's add allowedValues for the allowed values and set an explicit default

Reply:
I am not sure of allowed values. Could we set explicit default?

Reply:
Let's start with "http" and "https" then

Reply:
Removed this as discussed in call